framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -12,18 +12,278 @@
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+	): Promise<void> {
+		const { header, params: txParams } = context;
+		const chainIDBuffer = getIDAsKeyForStore(txParams.sendingChainID);
+		const partnerChainStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_DATA);
+		const partnerChainAccount = await partnerChainStore.getWithSchema<ChainAccount>(
+			chainIDBuffer,
+			chainAccountSchema,
+		);
+
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+
+		// if the CCU also contains a non-empty inboxUpdate, check the validity of certificate with liveness check
+		checkValidCertificateLiveness(txParams, header, decodedCertificate);
+
+		const partnerValidatorStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_VALIDATORS);
+		const partnerValidators = await partnerValidatorStore.getWithSchema<ChainValidators>(
+			chainIDBuffer,
+			chainValidatorsSchema,
+		);
+
+		// Certificate and Validators Update Validity
+		checkCertificateTimestampAndSignature(
+			txParams,
+			partnerValidators,
+			partnerChainAccount,
+			decodedCertificate,
+			header,
+		);
+
+		// If params contains a non-empty activeValidatorsUpdate
+		checkValidatorsHashWithCertificate(txParams, decodedCertificate, partnerValidators);
+
+		// CCM execution
+		const beforeSendContext = createCCMsgBeforeSendContext({
+			feeAddress: context.transaction.senderAddress,
+			eventQueue: context.eventQueue,
+			getAPIContext: context.getAPIContext,
+			logger: context.logger,
+			networkIdentifier: context.networkIdentifier,
+			getStore: context.getStore,
+		});
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		let decodedCCMs;
+		try {
+			decodedCCMs = txParams.inboxUpdate.crossChainMessages.map(ccm => ({
+				serialized: ccm,
+				deserialized: codec.decode<CCMsg>(ccmSchema, ccm),
+			}));
+		} catch (err) {
+			await interoperabilityStore.terminateChainInternal(
+				txParams.sendingChainID,
+				beforeSendContext,
+			);
+
+			throw err;
+		}
+		if (
+			partnerChainAccount.status === CHAIN_REGISTERED &&
+			!isInboxUpdateEmpty(txParams.inboxUpdate)
+		) {
+			// If the first CCM in inboxUpdate is a registration CCM
+			if (
+				decodedCCMs[0].deserialized.crossChainCommandID === CROSS_CHAIN_COMMAND_ID_REGISTRATION &&
+				decodedCCMs[0].deserialized.receivingChainID === MAINCHAIN_ID
+			) {
+				partnerChainAccount.status = CHAIN_ACTIVE;
+			} else {
+				await interoperabilityStore.terminateChainInternal(
+					txParams.sendingChainID,
+					beforeSendContext,
+				);
+
+				return; // Exit CCU processing
+			}
+		}
+
+		for (const ccm of decodedCCMs) {
+			if (txParams.sendingChainID !== ccm.deserialized.sendingChainID) {
+				await interoperabilityStore.terminateChainInternal(
+					txParams.sendingChainID,
+					beforeSendContext,
+				);
+
+				continue;
```

Comment:
Why `continue`, wouldn't it automatically ?
same for `try catch` block below.

Reply:
we `continue` because we want to skip the processing of CCM any further and terminate the chain. If I don't continue then it will call `appendToInboxTree` or `apply` later on

Reply:
Related to above comments, Clarifying with research